src/diff.rs

```diff
@@ -216,6 +216,13 @@ impl<Id: Hash + Eq + Copy, T: Default + AddAssign, W: Copy + PartialEq> GraphDif
         Ok(())
     }
 
+    /// # Safety
+    /// Does not check that the node IDs are valid (i.e. not marked as deleted).
+    /// CAUTION: This replaces existing diff edges by the provided ones.
+    pub unsafe fn set_edges_unchecked(&mut self, edges: HashMap<Id, HashMap<Id, W>>) {
+        let _ = std::mem::replace(&mut self.edges.new_or_updated, edges);
```

Comment:
What is the difference between
```
let _ = std::mem::replace(&mut self.a,b);
```
And
```
self.a = b;
```
In the first case we could return the old value but if we don't want to then is there anything else?

Reply:
- [ ] unnecessary replace